frontend/src/container/GeneralSettings/index.tsx

```diff
@@ -5,37 +5,66 @@ import Spinner from 'components/Spinner';
 import React from 'react';
 import { useTranslation } from 'react-i18next';
 import { useQueries } from 'react-query';
+import { ErrorResponse, SuccessResponse } from 'types/api';
+import { PayloadProps as GetRetentionPeriodAPIPayloadProps } from 'types/api/settings/getRetention';
 
 import GeneralSettingsContainer from './GeneralSettings';
 
 function GeneralSettings(): JSX.Element {
 	const { t } = useTranslation('common');
-	const [getRetentionPeriodApiResponse, getDisksResponse] = useQueries([
+	const [
+		getRetentionPeriodMetricsApiResponse,
+		getRetentionPeriodTracesApiResponse,
+		getDisksResponse,
+	] = useQueries([
 		{
-			queryFn: getRetentionPeriodApi,
-			queryKey: 'getRetentionPeriodApi',
+			queryFn: (): Promise<
```

Comment:
Can we make a common type here as we are changing metrics and traces only

Reply:
@pranshuchittora ^^